### ISSUE TITLE
releases and versionCode

### ISSUE THREAD

**Issue by IzzySoft** [hjiangsu/thunder#10]
Please remember to increase `versionCode` whenever you make a new release. Currently, it's still set to `1` despite of there being more than one release. Android uses `versionCode` to tell versions apart – so right now, to Android both your releases are the very same version; which results in the update not being signaled as it thinks it's already installed.

Hint: it's [this line](https://github.com/hjiangsu/thunder/blob/main/pubspec.yaml#L4). Here's how it should currently look instead:

```yaml
version: 0.1.1-alpha+2
```

The `+2` is telling Flutter to use that as `versionCode`. If not set (i.e. there is no `+<integer>` part), it defaults to `1`.

Thanks for taking care – and all the best for your project!

**Comment by hjiangsu** [comment]
Hey, thanks for letting me know! I did change the version number, but maybe there's something I'm doing incorrectly? (maybe you can help me understand better)

The initial version which had the internet permission issue was `0.1.0-alpha`, and when I added in the fix (plus a few other additions), I changed the version to `0.1.1-alpha`. Would this be correct?

**Comment by IzzySoft** [comment]
What you change becomes the `versionName`. What I mean is the `versionCode` – which seems to be left out with the Flutter templates, and Flutter itself defaults it to `1`. In order to fix that, you **append** it to your `version:`, using a `+` as separator. Noticed my above string ends with `+2`? That tells Flutter to not use the default (`1`), but what is specified here (`2`). Always increase that number whenever you make a new release. Increasing by `1` is fine, any larger number is also fine – it just has to be an integer larger than the last one.

As for the `versionName` (which is what you refer to), you are totally free to make it whatever you want. Yes, what you outlined is totally fine, even matches [semantic versioning](https://semver.org/) which is a very good practice.

TL;DR:

```yaml
version: <versionName>[+<versionCode>]
```

You need to add the optional part :wink:

**Comment by hjiangsu** [comment]
Ahh, thanks so much for explaining it! That makes a lot more sense now, I'll make sure to do this in the future releases

**Comment by hjiangsu** [comment]
Closing issue

**Comment by IzzySoft** [comment]
Thanks!

**Comment by IzzySoft** [comment]
Looks good, thanks:

```
$ iod repo get com.hjiangsu.thunder
com.hjiangsu.thunder: looking for 'https://api.github.com/repos/hjiangsu/thunder/releases'
com.hjiangsu.thunder: checking tag 'v0.2.0-alpha'
com.hjiangsu.thunder: lastRelNo set to '0.2.0-alpha', checking for files
com.hjiangsu.thunder: Upstream file date (2023-06-20 17:21) is newer than ours (2023-06-17 05:01).
com.hjiangsu.thunder: returning ['0.2.0-alpha','https://github.com/hjiangsu/thunder/releases/download/v0.2.0-alpha/thunder-v0.2.0-alpha.apk',1687274488]
com.hjiangsu.thunder: 0.1.1-alpha/0.2.0-alpha, https://github.com/hjiangsu/thunder/releases: https://github.com/hjiangsu/thunder/releases/download/v0.2.0-alpha/thunder-v0.2.0-alpha.apk
- Grabbing update for com.hjiangsu.thunder: OK
- Checking 'repo/com.hjiangsu.thunder_3.apk' for libraries and malware …
com.hjiangsu.thunder: check if repo contains FUNDING.yml
com.hjiangsu.thunder: looking for 'https://api.github.com/repos/hjiangsu/thunder/contents/.github'
com.hjiangsu.thunder: looking for 'https://api.github.com/repos/hjiangsu/thunder/contents/'
com.hjiangsu.thunder: no FUNDING.yml detected.
com.hjiangsu.thunder: no Fastlane configured, skipping Fastlane check.
$ _
```

As for Fastlane: should you set up its metadata (i.e. app description and graphics), my updater would pull that along as well. My [Fastlane Cheat Sheet](https://gitlab.com/-/snippets/1895688) can guide you with that if you wish.

**Comment by hjiangsu** [comment]
Awesome, glad to hear that! Yeah, I just started to mess around with Fastlane, but haven't done too much with it yet - thanks for the links!

**Comment by IzzySoft** [comment]
Want a PR with a "starter package", containing what is currently set up with my repo? If so: I found you have a `fastlane` folder in some subdirectory. That works with my repo (as I can flexibly configure the entry point) but won't work with F-Droid.org (which expects it in the repo root), so let me know if I shall feed the existing one (you can always move it later when needed) or add it in the correct place.

**Comment by hjiangsu** [comment]
> Want a PR with a "starter package", containing what is currently set up with my repo?

That would be much appreciated!

>  but won't work with F-Droid.org (which expects it in the repo root)

I'm not too familiar myself with how F-Droid works, but for now, I would say its okay to just leave it in the existing directories (the reason why it is currently in a subdirectory rather than the root directory is to allow separate fastlane "configs" for iOS and Android according to https://docs.flutter.dev/deployment/cd#fastlane)

**Comment by IzzySoft** [comment]
OK, there you go!

While on it: I just had to add the `Tracking` anti-feature to Thunder after yesterday's update. My library scanner rang the alert bell, having detected Sentry – which is not bad per se, but I found it's enabled by default and not even a switch in the settings to turn it off. To be able to remove the AF, Sentry would need to be configured opt-in (e.g. ask directly at first start whether it's OK to enable it, and give the option to change ones mind in the settings screen).

**Comment by hjiangsu** [comment]
Thanks! And for sure, I'll definitely make Sentry an opt-in feature in the next release. I just wanted to quickly push that out for now to address any immediate issues that pop up due to a release since there's been so much traction here.

**Comment by IzzySoft** [comment]
OK, got it – thanks! Please let me know when the release with it being opt-in has made it to my repo, so I can remove the AF again.

**Comment by IzzySoft** [comment]
Manually triggered the update and removed the AF, as promised. Thanks! Btw, Fastlane came along fine:

```
$ iod repo get com.hjiangsu.thunder
com.hjiangsu.thunder: looking for 'https://api.github.com/repos/hjiangsu/thunder/releases'
com.hjiangsu.thunder: checking tag 'v0.2.1+5'
com.hjiangsu.thunder: lastRelNo set to '0.2.1+5', checking for files
com.hjiangsu.thunder: Upstream file date (2023-06-22 21:24) is newer than ours (2023-06-20 19:15).
com.hjiangsu.thunder: returning ['0.2.1+5','https://github.com/hjiangsu/thunder/releases/download/v0.2.1%2B5/thunder-v0.2.1%2B5.apk',1687461866]
com.hjiangsu.thunder: 0.2.0-alpha/0.2.1+5, https://github.com/hjiangsu/thunder/releases: https://github.com/hjiangsu/thunder/releases/download/v0.2.1%2B5/thunder-v0.2.1%2B5.apk
- Grabbing update for com.hjiangsu.thunder: OK
- Checking 'repo/com.hjiangsu.thunder_5.apk' for libraries and malware …
com.hjiangsu.thunder: check if repo contains FUNDING.yml
com.hjiangsu.thunder: looking for 'https://api.github.com/repos/hjiangsu/thunder/contents/.github'
com.hjiangsu.thunder: looking for 'https://api.github.com/repos/hjiangsu/thunder/contents/'
com.hjiangsu.thunder: no FUNDING.yml detected.
com.hjiangsu.thunder: calling 'getFastlaneMeta(github,[host:github.com,owner:hjiangsu,repo:thunder,path:/android/fastlane/metadata/android])'
com.hjiangsu.thunder: FastlaneFeatures shortdesc,fulldesc,icon,screenshotsJPG
com.hjiangsu.thunder: looking for 'https://api.github.com/repos/hjiangsu/thunder/contents/android%2Ffastlane%2Fmetadata%2Fandroid'
com.hjiangsu.thunder: looking for 'https://api.github.com/repos/hjiangsu/thunder/contents/android%2Ffastlane%2Fmetadata%2Fandroid%2Fen-US'
com.hjiangsu.thunder: looking for 'https://api.github.com/repos/hjiangsu/thunder/contents/android%2Ffastlane%2Fmetadata%2Fandroid%2Fen-US%2Fimages'
com.hjiangsu.thunder: looking for 'https://api.github.com/repos/hjiangsu/thunder/contents/android%2Ffastlane%2Fmetadata%2Fandroid%2Fen-US%2Fimages%2FphoneScreenshots'
com.hjiangsu.thunder: checking locale 'en-US'
com.hjiangsu.thunder: updating 'metadata/com.hjiangsu.thunder/en-US/full_description.txt'
com.hjiangsu.thunder: updating 'metadata/com.hjiangsu.thunder/en-US/short_description.txt'
com.hjiangsu.thunder: updating 'repo/com.hjiangsu.thunder/en-US/icon.png'
com.hjiangsu.thunder: updating 'repo/com.hjiangsu.thunder/en-US/phoneScreenshots/screenshot_1.jpg' as JPG
com.hjiangsu.thunder: updating 'repo/com.hjiangsu.thunder/en-US/phoneScreenshots/screenshot_2.jpg' as JPG
com.hjiangsu.thunder: updating 'repo/com.hjiangsu.thunder/en-US/phoneScreenshots/screenshot_3.jpg' as JPG
com.hjiangsu.thunder: cross-checking for obsolete screenshots
com.hjiangsu.thunder: screenshots in Fastlane: screenshot_1,screenshot_2,screenshot_3
com.hjiangsu.thunder: local screenshots checked: screenshot_1,screenshot_2,screenshot_3
$ _
```
